### ISSUE TITLE
Statements failed in pylance but passed in mypy

### ISSUE THREAD

**Issue by TheNightmareX** [microsoft/pylance-release#861]
## Environment data

-   Language Server version: 2021.1.1
-   OS and version: Windows 10 Work Station 1909
-   Python version: Python 3.9.0 64-bit

## Actual behaviour

```Python
from typing import Dict, List, Type


class BaseClass:
    ...


class SubClass(BaseClass):
    ...


# these are ok
cls: Type[BaseClass] = SubClass
cls_dict: Dict[str, Type[BaseClass]] = {'string': SubClass}
cls_list: List[Type[BaseClass]] = [SubClass]


class A:
    # these are ok
    cls: Type[BaseClass] = SubClass
    cls_dict: Dict[str, Type[BaseClass]] = {'string': SubClass}
    cls_list: List[Type[BaseClass]] = [SubClass]


class B(A):
    # this is ok
    cls = SubClass

    # but the following statements failed in pylance but passed in mypy

    cls_dict = {'string': SubClass}
    """
    (variable) cls_dict: dict[str, Type[SubClass]]
    "cls_dict" overrides symbol of same name in class "A"
    TypeVar "_VT@dict" is invariant
        "SubClass" is incompatible with "BaseClass"PylancereportIncompatibleVariableOverride
    test.py(11, 5): Overridden symbol
    """

    cls_list = [SubClass]
    """
    (variable) cls_list: list[Type[SubClass]]
    "cls_list" overrides symbol of same name in class "A"
    TypeVar "_T@list" is invariant
        "SubClass" is incompatible with "BaseClass"PylancereportIncompatibleVariableOverride
    test.py(12, 5): Overridden symbol
    """
```

**Comment by judej** [comment]
@erictraut type checking issue in class scope?

**Comment by erictraut** [comment]
This looks like it's a bug in the `reportIncompatibleVariableOverride` diagnostic rule. I'll take a look.

**Comment by erictraut** [comment]
The bug is particular to this specific case, which is why we haven't seen it before. It requires that:
1. A base class has a class variable
2. There is a type declared for that class variable
3. A subclass has the same class variable but does not provide a declared type for it
4. The subclass assigns a value to that class variable
5. The inferred type of the assigned expression depends on the declared type of the class variable

I've address the problem in this PR: https://github.com/microsoft/pyright/commit/c92c3ef66a15e039ff522103fe01c2e615c15931.

It will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.1.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202113-27-january-2021
